### PR TITLE
Extend CMakeLists.txt to generate a package

### PR DESCRIPTION
**Pull request by ThijsWithaar**
This extends the CMake script to generate a .deb package under debian.
It allows other projects to use find_package and remove this library using the system's package manager.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,13 @@
 cmake_minimum_required (VERSION 3.5) # The minimum version of CMake necessary to build this project
 
+set( TRISYCL_VERSION_MAJOR 0)
```
Comment (keryell) [inline CMakeLists.txt]: It makes me think that we do not have a versioning plan for triSYCL...
But your 0.1.0 looks like a good start.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+set( TRISYCL_VERSION_MINOR 1)
+set( TRISYCL_VERSION_PATCH 0)
+set( TRISYCL_VERSION "${TRISYCL_VERSION_MAJOR}.${TRISYCL_VERSION_MINOR}.${TRISYCL_VERSION_PATCH}")
```
Comment (keryell) [inline CMakeLists.txt]: I guess that the spaces or not inside the '()` should be the same for all the lines of the file?

File: CMakeLists.txt [hunk 3]
```diff
@@ -14,3 +22,74 @@ include_directories(${PROJECT_SOURCE_DIR}/tests/common) # All targets inherit SY
...
+
+
+# CPACK
+
+set( CPACK_GENERATOR "TGZ;DEB" )
+set( CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}-${TRISYCL_VERSION}-${CMAKE_HOST_SYSTEM_NAME}" )
+set( CPACK_SOURCE_PACKAGE_FILE_NAME "${CPACK_PACKAGE_FILE_NAME}-Source" )
+
+set( CPACK_PACKAGE_CONTACT "${PROJECT_NAME} Maintainers")
+set( CPACK_PACKAGE_DESCRIPTION_SUMMARY "SyCL reference implementation" )
```
Comment (keryell) [inline CMakeLists.txt]: s/SyCL/SYCL/

File: CMakeLists.txt [hunk 4]
```diff
@@ -14,3 +22,74 @@ include_directories(${PROJECT_SOURCE_DIR}/tests/common) # All targets inherit SY
...
+set( CPACK_PACKAGE_NAME ${PROJECT_NAME})
+set( CPACK_PACKAGE_VENDOR "Neutral" )
+set( CPACK_RESOURCE_FILE_README "${CMAKE_SOURCE_DIR}/README.rst")
+
+set( CPACK_PACKAGE_VERSION_MAJOR ${TRISYCL_VERSION_MAJOR} )
+set( CPACK_PACKAGE_VERSION_MINOR ${TRISYCL_VERSION_MINOR} )
+set( CPACK_PACKAGE_VERSION_PATCH ${TRISYCL_VERSION_PATCH} )
+
+set( CPACK_DEBIAN_PACKAGE_DEPENDS "opencl-c-headers (>=1.2)")
```
Comment (keryell) [inline CMakeLists.txt]: There is probably a need to add somewhere some dependencies, such as Boost.Compute and others.

**Comment by ThijsWithaar** [comment]
I think I have covered your feedback.
It does install on my system, were 3rd party projects can do find_package() and use 'triSYCL::triSYCL' in their target_include_directories.

Please let me know if there's anything else to cover.

**Pull request merged**